skimage/measure/__init__.py

```diff
@@ -7,7 +7,7 @@
 from ._pnpoly import points_in_poly, grid_points_in_poly
 from ._moments import moments, moments_central, moments_normalized, moments_hu
 from .profile import profile_line
-from .fit import LineModel, CircleModel, EllipseModel, ransac
+from .fit import LineModel, LineModel3D, CircleModel, EllipseModel, ransac
```

Comment:
Hi,

As the functions estimate() and residuals() of LineModel3D are not 3D
specific, they would work in 2D, 4D or beyond, I see how to write a single
LineModel that only exposes the old params (offset, theta) and predict_x()
and predict_y() in the 2D case, checking the dimensionality of the data as
you suggest. I will do it tomorrow and update the pull request.

Kind Regards,

Kevin
On 1 Dec 2015 17:01, "Josh Warner" <notifications@github.com> wrote:

> In skimage/measure/__init__.py
> <https://github.com/scikit-image/scikit-image/pull/1795#discussion_r46305514>
> :
>
> > @@ -7,7 +7,7 @@
> >  from ._pnpoly import points_in_poly, grid_points_in_poly
> >  from ._moments import moments, moments_central, moments_normalized, moments_hu
> >  from .profile import profile_line
> > -from .fit import LineModel, CircleModel, EllipseModel, ransac
> > +from .fit import LineModel, LineModel3D, CircleModel, EllipseModel, ransac
>
> Hmm. It would be simple enough to harmonize the outputs to (point, unit
> vector) - potentially adding an attribute (old_params?) or kwarg to
> regain the old behavior. In pursuit of a dramatically improved,
> generalizable, API I would support a breaking change.
>
> Regarding other methods, it's easy enough to house all logic for this in a ._check(self,
> data) method called from estiamate or residuals and populate the rest of
> the class, including optional methods. Particularly because it seems
> predict_x and predict_y are intended to be called after estimate.
> Alternatively, LineModel could simply not expose them but LineModel2D
> could retain them.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/scikit-image/scikit-image/pull/1795/files#r46305514>.
>